### PR TITLE
Ansible minio spark

### PR DESCRIPTION
**Pull request by romulogoncalves**
We use Spark without Hadoop. To have hadoop we install Hadoop 2.8.0 and tell Spark to use it.
Like this we can use hadoop-aws 2.8.0 which works with aws-java-sdk-s3-1.10.6. With it we can read and process data from Minion using S3A interface. At the moment we write the results to HDFS, this is, we only use S3 to read to data like it was a large storage.

An issue is still open waiting for a solution which allow us to write to Minio.

In the roles for spark and hadoop the tasks to start and stop the services are now part of independent playbooks. Like this I only need to install and set the cluster once. Then everytime the machines are up we just need to start the services (name node, data nodes, hdfs, spark-master and spark-workers).

### CODE REVIEW

File: roles/common/tasks/main.yml [hunk 1]
```diff
@@ -41,18 +48,8 @@
...
-  ignore_errors: true
-  register: restarted
-- name: Waiting for reboot...
-  local_action: wait_for host={{ ansible_ssh_host | default(inventory_hostname) }} search_regex=OpenSSH port=22 delay=15 state=started
-  sudo: no
-  when: restarted.changed
-- name: Rebooting machine so new kernel is active
-  command: shutdown -r now removes=/var/run/reboot-required
-  async: 0
+  command: sleep5 && shutdown -r now removes=/var/run/reboot-required
```
Comment (sverhoeven) [inline roles/common/tasks/main.yml]: Space missing

File: roles/hadoop/templates/hadoop-hdfs.systemd.j2 [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=hadoop-hdfs
+After=syslog.target network.target
+
+[Service]
+User=ubuntu
+Type=forking
+Environment="JAVA_HOME=/usr/lib/jvm/java-8-oracle"
```
Comment (sverhoeven) [inline roles/hadoop/templates/hadoop-hdfs.systemd.j2]: Use ansible variable, instead of hardcoding

File: roles/hadoop/files/id_rsa [hunk 3]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (sverhoeven) [inline roles/hadoop/files/id_rsa]: Please remove and add id_rsa, id_rsa.pub to .gitignore to prevent re-adding

Comment (romulogoncalves) [inline roles/hadoop/files/id_rsa]: Fixed in:
https://github.com/nlesc-sherlock/emma/commit/1a56f6d1e3fbb73d045b8f137e1239dd440c540f

File: roles/common/tasks/main.yml (hunk 1 above)
Comment (romulogoncalves) [inline roles/common/tasks/main.yml]: Fixed in:
https://github.com/nlesc-sherlock/emma/commit/f1f4a6ad82374a69e7d6e505136fc850009d7742

File: roles/hadoop/templates/hadoop-hdfs.systemd.j2 (hunk 2 above)
Comment (romulogoncalves) [inline roles/hadoop/templates/hadoop-hdfs.systemd.j2]: Fixed in:
https://github.com/nlesc-sherlock/emma/commit/b77a04dc06bc76b725007dc1ea0482bbce02cc60
and:
https://github.com/nlesc-sherlock/emma/commit/854de7b8005299d9875a2999bd6e9838d9c71982

**Pull request merged**